### PR TITLE
Issue 1269: annotation option

### PR DESCRIPTION
**Pull request by kostrubin**
Annotation type option `@FindBy` was added.

### CODE REVIEW

File: src/features/locators/Locator.tsx [hunk 1]
```diff
@@ -131,8 +137,11 @@ export const Locator: React.FC<Props> = ({ element, currentPage, searchState, de
 
     return (
       <span onClick={handleClick}>
-        @UI(
-        <span className="jdn__xpath_item-locator">&quot;{locator.output}&quot;</span>)
+        {annotationType}(
+        <span className="jdn__xpath_item-locator">
+          {isUiAnnotationType() ? "" : isXpathLocatorType() ? "xpath = " : "css = "}&quot;{locator.output}&quot;
```
Comment (vitaepam) [inline src/features/locators/Locator.tsx:142]: Maybe while you update this code, move texts to constants?

**Review by vitaepam — APPROVED** [review]

File: src/features/locators/Locator.tsx (hunk 1 above)
Comment (kostrubin) [reply src/features/locators/Locator.tsx]: Moved it to a separate helper function, not sure if it’s really needed in this case 🙂

File: src/features/locators/types/locator.types.ts [hunk 2]
```diff
@@ -95,7 +95,8 @@ export interface Locator extends PredictedEntity {
   active?: boolean;
   isCustomName?: boolean;
   isCustomLocator?: boolean;
-  locatorType?: LocatorType;
+  annotationType: AnnotationType;
+  locatorType: LocatorType;
```
Comment (MariiaNebesnova) [inline src/features/locators/types/locator.types.ts:99]: In fact, locator doesn't have this attribute until user set manually it in Edit Form. Changing it to requeried will brake the logic in several places

File: src/features/locators/components/LocatorEditDialog.tsx [hunk 3]
```diff
@@ -293,6 +300,25 @@ export const LocatorEditDialog: React.FC<Props> = ({
           ]}
         />
       </FormItem>
+      <FormItem
+        wrapperCol={{ span: 24, xs: { offset: 0 }, sm: { offset: 4 } }}
+        name="annotationType"
+        style={{ marginBottom: "8px" }}
+      >
+        <Select
+          disabled={!isValidLocator(validationMessage)}
```
Comment (arslanbekova) [inline src/features/locators/components/LocatorEditDialog.tsx:309]: I couldn't find a requirement about disabling this select in ticket. Should we do it?

Comment (kostrubin) [reply src/features/locators/components/LocatorEditDialog.tsx]: Thanks for pointing that out, removed

File: src/features/locators/types/locator.types.ts (hunk 2 above)
Comment (kostrubin) [reply src/features/locators/types/locator.types.ts]: Rolled back

File: src/common/utils/helpers.ts [hunk 4]
```diff
@@ -1,14 +1,31 @@
...
+export const getLocatorString = (
+  annotationType: AnnotationType | undefined,
+  locatorType: LocatorType | undefined,
+  locator: LocatorValue,
+  type: ElementLibrary | ElementClass,
+  name: string
+): string =>
+  `${annotationType || AnnotationType.UI}(${getLocatorPrefix(annotationType, locatorType)}"${
+    locator.output
+  }")\npublic ${type} ${name};`;
```
Comment (MariiaNebesnova) [inline src/common/utils/helpers.ts:28]: refact location of these functions, please. move them to features/locators/utils/locatorOutput.ts

File: src/features/locators/utils/utils.ts [hunk 5]
```diff
@@ -15,14 +15,15 @@ import {
...
 import { isStringContainsNumbers } from "../../../common/utils/helpers";
 import { FormInstance } from "antd/es/form/Form";
 import { copyToClipboard } from "../../../common/utils/copyToClipboard";
 
-export const getLocatorWithJDIAnnotation = (locator: string): string => `@UI("${locator}")`;
+export const getLocatorWithJDIAnnotation = (locator: string): string => `${AnnotationType.UI}}("${locator}")`;
 
-export const getLocatorWithSelenium = (locator: string, option: string): string => `@FindBy(${option} = "${locator}")`;
+export const getLocatorWithSelenium = (locator: string, option: string): string =>
+  `${AnnotationType.FindBy}(${option} = "${locator}")`;
```
Comment (MariiaNebesnova) [inline src/features/locators/utils/utils.ts:26]: please, move them to features/locators/utils/locatorOutput.ts as well

**Review by MariiaNebesnova — APPROVED** [review]
Thank you for code! merge, please

**Pull request merged**